tools/sign_executable/sign_executable.py

```diff
@@ -43,103 +40,48 @@
-            os.path.join(executable_name, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib"),
-        ]
-
-        for dir, dylib in zip(add_rpath_dirs, add_rpath_dylibs):
-            cmd = ["install_name_tool", "-add_rpath", dir, dylib]
-            spear.log(f"Executing: {' '.join(cmd)}")
-            subprocess.run(cmd, check=True)
-
-        # revert current working directory
-        spear.log(f"Changing working directory to {cwd}")
-        os.chdir(cwd)
-
-        # files that need to be code-signed
-        sign_files = [
-            os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Intel", "TBB", "Mac", "libtbb.dylib")),
-            os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Intel", "TBB", "Mac", "libtbbmalloc.dylib")),
-            os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac", "libogg.dylib")),
-            os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib")),
-            os.path.realpath(os.path.join(executable, "Contents", "MacOS", os.path.splitext(executable_name)[0]))
-        ]
-
-        assert os.path.exists(args.entitlements_file)
-
-        for file in sign_files:
-            cmd = [
-                "sudo", "codesign", "-f", "-s", "-v", "--options", "runtime", "--timestamp", "--entitlements", 
-                args.entitlements_file, "--sign", f"Developer ID Application: {args.developer_id}", file]
-            spear.log(f"Executing: {' '.join(cmd)}")
-            subprocess.run(cmd, check=True)
-
-        # create a zip file for notarization
-        notarization_zip = os.path.realpath(os.path.join(args.temp_dir, f"{os.path.splitext(executable_name)[0]}.zip"))
-        cmd = ["ditto", "-c", "-k", "--rsrc", "--keepParent", executable, notarization_zip]
-        spear.log(f"Executing: {' '.join(cmd)}")
-        subprocess.run(cmd, check=True)
+    executable_name = os.path.basename(executable)
+
+    # files that need to be code-signed
+    sign_files = [
+        os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Intel", "TBB", "Mac", "libtbb.dylib")),
+        os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Intel", "TBB", "Mac", "libtbbmalloc.dylib")),
+        os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac", "libogg.dylib")),
+        os.path.realpath(os.path.join(executable, "Contents", "UE", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib")),
+        os.path.realpath(os.path.join(executable, "Contents", "MacOS", os.path.splitext(executable_name)[0]))
+    ]
 
-        # send the zip file for notarization
+    assert os.path.exists(args.entitlements_file)
```

Comment:
Let's move this `assert` right below `parse_args()` because asserting earlier is better